### DISCUSSION TITLE
Palettes From Images

### DISCUSSION THREAD

**Discussion by DWrightAnimation** [keiwando/toonsquid#710] [Ideas]
I don't know how I didn't notice this feature all this time, probably because I rarely use the palette feature in general, but Procreate has the ability to extract a color palette from an image, as well as from a file and a picture from the camera. Unfortunately, I can't import the palettes from Procreate into ToonSquid, and vice versa. I know this feature isn't planned for the 2.0 update, but it'd be nice to have this feature in ToonSquid to extract colors from character art, concept art, etc.

**Comment by Cat-Butler** [comment]
Best of the season's wishes to everybody! 
Heri za Kwanzaa, Eid Mubarak, Hanukkah Sameach, Celebratory Dies Solis Invicti Nati and/or Joyous Mithra's Day, Merry Christmas, Festive Shab-e Yalda, Happy Haloa, Blessed Solstice and/or Convivial Saturnalia, Radiant Ra-ness, Jubilant Yule, Jovial Santa Season, Exuberant Family Gathering, Buoyant Break, or whatever you reflect on and/or celebrate (or not) at this season of the year.

As a gift and celebration for the season, here's a converter for everybody that can run on your iPad, for converting Procreate palettes (.swatches) to ToonSquid (.palette). Works on both Procreate version 4 (converts from HSB data) and version 5 palette formats (reads RGB components directly).
Written in JavaScript. You can open it and check the (somewhat amateur, quick-and-dirty) source in any text editor.
(Note: will also run on desktop computers, probably on any capable web browser)

[Google Drive link to download pc2ts.html](https://drive.google.com/file/d/1VUr-1XRvaYmnclbv2j04a2TaSHIC8g5B/view?usp=drive_link)

Conditions:
You will probably need to install the Edge browser (because it automatically sets itself up as a Share target on iPadOS, which makes the whole thing a lot easier)
You will need to be online when you convert (because every time the file loads in the browser, it then grabs an online zip handling library and online file saving library, for convenience)

- Download the (HTML) file somewhere locally
- Share > More > Edge, to open it as a local web page.
- Hit the Choose File button on the web page, Choose File from the popup, choose a .gpl file.
- Choose the Download option for the generated .palette file, then Show, then use the 3-vertical-dot (kebab) menu to Share the downloaded .palette file, Save to Files.
- Import it in ToonSquid as a palette - Palettes tab in the colour panel, 3-horizontal-dots (meatballs) menu, Import.

Here's an example video of it in action, and showing the resulting palette in ToonSquid:

https://github.com/user-attachments/assets/ced71ed0-97ef-4659-bde5-170dcaad3e25

Note: some third-party version 5 Procreate palettes have a flaw where the internal RGB components do not match the HSB data - I've run into a few of these. If you see a ToonSquid palette that's not what you were expecting, often with way too much blue, use this converter instead (otherwise identical), which ignores the usual version 5 palette's RGB components and converts the internal HSB data instead (the same as a version 4 palettes):

[pc2ts-broken.html](https://drive.google.com/file/d/1Xgyx0nxhzhnxaCY1AkFN7uF3rSB8vtz7/view?usp=drive_link)

Other palette formats are left as an exercise for the reader.
